test/e2e-go/features/devmode/devmode_test.go

```diff
@@ -75,3 +75,39 @@ func TestDevMode(t *testing.T) {
 		prevTime = currTime
 	}
 }
+
+// Starts up a devmode network, sends a txn, and fetches the txn group delta for that txn
+func TestTxnGroupDeltasDevMode(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	if testing.Short() {
+		t.Skip()
+	}
+
+	// Start devmode network, and send a transaction.
+	var fixture fixtures.RestClientFixture
+	fixture.SetupNoStart(t, filepath.Join("nettemplates", "DevModeTxnTracerNetwork.json"))
+	fixture.Start()
+	defer fixture.Shutdown()
+	sender, err := fixture.GetRichestAccount()
+	require.NoError(t, err)
+	key := crypto.GenerateSignatureSecrets(crypto.Seed{})
+	receiver := basics.Address(key.SignatureVerifier)
+	txn := fixture.SendMoneyAndWait(0, 100000, 1000, sender.Address, receiver.String(), "")
+	require.NotNil(t, txn.ConfirmedRound)
+	_, err = fixture.AlgodClient.Block(*txn.ConfirmedRound)
+	require.NoError(t, err)
+
+	// Test GetLedgerStateDeltaForTransactionGroup and verify the response contains a delta
+	txngroupResponse, err := fixture.AlgodClient.GetLedgerStateDeltaForTransactionGroup(txn.Txn.ID().String())
+	require.NoError(t, err)
```

Comment:
When I ran the test locally _with the old code_ it failed showing that the changes in `ledger.go` are a good fix. However, it failed at line 103 and we never got to assert in line 104.

This begs the following question: is it possible to get a non-404 yet still have `len(txngroupResponse) == 0` ?

Reply:
The length should be 0 if the call errors. This API can return `400`, `404`, `500`, `501` error codes. All of those would populate the error field since we're only counting `200` or `201` as successful responses in our tests here.